src/EventStore.Core.Tests/Http/Streams/when_posting_events_with_expected_versions.cs

```diff
@@ -0,0 +1,287 @@
+            request.ContentLength = data.Length;
+            request.GetRequestStream().Write(data, 0, data.Length);
+            return GetRequestResponse(request);
+        }
+    }
+
+    [TestFixture, Category("LongRunning")]
+    public class when_posting_event_with_expected_version_any_to_new_stream : ExpectedVersionSpecification
+    {
+        private HttpWebResponse _response;
+
+        protected override void Given()
+        {
+        }
+
+        protected override void When()
+        {
+            _response = PostEventWithExpectedVersion(ExpectedVersion.Any);
+        }
+
+        [Test]
+        public void request_succeeds()
+        {
+            Assert.AreEqual(HttpStatusCode.Created, _response.StatusCode);
+        }
+    }
+
+    [TestFixture, Category("LongRunning")]
+    public class when_posting_event_with_expectedVersion_no_stream_to_new_stream : ExpectedVersionSpecification
+    {
+        private HttpWebResponse _response;
+
+        protected override void Given()
+        {
+        }
+
+        protected override void When()
+        {
+            _response = PostEventWithExpectedVersion(ExpectedVersion.NoStream);
+        }
+
+        [Test]
+        public void request_succeeds()
+        {
+            Assert.AreEqual(HttpStatusCode.Created, _response.StatusCode);
+        }
+    }
+
+    [TestFixture, Category("LongRunning")]
+    public class when_posting_event_with_expectedVersion_no_stream_to_existing_stream : ExpectedVersionSpecification
+    {
+        private HttpWebResponse _response;
+
+        protected override void Given()
+        {
+            PostEventWithExpectedVersion(ExpectedVersion.Any);
+        }
+
+        protected override void When()
+        {
+            _response = PostEventWithExpectedVersion(ExpectedVersion.NoStream);
+        }
+
+        [Test]
+        public void request_fails_with_wrong_event_number()
+        {
+            Assert.AreEqual(HttpStatusCode.BadRequest, _response.StatusCode);
+            Assert.AreEqual(WrongExpectedVersionDesc, _response.StatusDescription);
+        }
+    }
+
+    [TestFixture, Category("LongRunning")]
+    public class when_posting_event_with_expectedVersion_of_an_event_position_to_new_stream : ExpectedVersionSpecification
+    {
+        private HttpWebResponse _response;
+
+        protected override void Given()
+        {
+        }
+
+        protected override void When()
+        {
+            _response = PostEventWithExpectedVersion(1);
+        }
+
+        [Test]
+        public void request_fails_with_wrong_event_number()
+        {
+            Assert.AreEqual(HttpStatusCode.BadRequest, _response.StatusCode);
+            Assert.AreEqual(WrongExpectedVersionDesc, _response.StatusDescription);
+        }
+    }
+
+    [TestFixture, Category("LongRunning")]
+    public class when_posting_event_with_expected_version_higher_than_the_number_of_events : ExpectedVersionSpecification
```

Comment:
It might be a good idea to keep these tests consistent with regards to naming and expectancies to those of the ClientAPI for example https://github.com/EventStore/EventStore/blob/699c17fc01ab633ceb14fac7a6e8ecd2c5535c18/src/EventStore.Core.Tests/ClientAPI/append_to_stream.cs#L259.